lib/srv/regular/sshserver.go

```diff
@@ -281,6 +301,11 @@ func (s *Server) close() {
 	if s.dynamicLabels != nil {
 		s.dynamicLabels.Close()
 	}
+
+	if s.users != nil {
+		s.users.Shutdown()
```

Comment:
> one thing I'm unclear on is the best way to guarantee the semaphore gets deleted after a service stops, is there a standard way in teleport to watch for signals?

This should already be called on any shutdown signal, grep for `process.OnExit("ssh.shutdown"`, you should be able to track it down from there. Test it yourself too, of course.